waiter/src/waiter/core.clj

```diff
@@ -266,6 +266,48 @@
       (catch Exception e
         (utils/exception->response e request)))))
 
+(defn add-grpc-headers-and-trailers
+  "Finds and attaches the equivalent grpc status codes for the provided http status code."
+  [server-name {:keys [body headers status trailers] :as response}]
+  (if (= server-name (get-in response [:headers "server"]))
+    (if-let [grpc-status-data (cond
+                                (= status 400) ["3" "Bad Request"]
+                                (= status 401) ["16" "Unauthorized"]
+                                (= status 500) ["13" "Internal Server Error"]
+                                (= status 503) ["14" "Service Unavailable"]
+                                (= status 504) ["4" "Gateway Timeout"])]
+      (let [[grpc-status grpc-message] grpc-status-data
+            grpc-message (if (string? body)
+                           (utils/truncate body 128)
```

Comment:
Why are we truncating to 128 bytes?

Reply:
Removed the truncation. I verified that I could send 1000s of characters in the status message.